### PR TITLE
Convert SPI intel driver to DTS

### PR DESCRIPTION
**Pull request by tbursztyka**

### CODE REVIEW

**Comment by tbursztyka** [comment]
There is a chicken & egg issue: if I want to remove SPI_0_NAME from drivers/spi/Kconfig, I should first make cc2520 (ieee802154) dts ready. Will do that, and make this current PR on top of that 15.4 change branch.

**Comment by tbursztyka** [comment]
myeah there is even more SPI name config issue all over the samples, which means more cleanup there.

File: dts/bindings/spi/intel,intel-spi.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+description: >
+     This is a representation of the Intel spi node
+
+inherits:
+    !include spi.yaml
+
+properties:
+    compatible:
+      constraint: "intel,intel-spi"
```
Comment (galak) [inline dts/bindings/spi/intel,intel-spi.yaml]: Is there anything more detailed or descriptive than 'intel,intel-spi'?  Seems a bit too generic

**Review by galak — CHANGES_REQUESTED** [review]
Looks good beyond the compatible feeling a bit too generic.

**Comment by galak** [comment]
> myeah there is even more SPI name config issue all over the samples, which means more cleanup there.

Yeah, there is Issue #10894 that covers all the SPI slave drivers that need converting over.  Let's get the controller stuff merged and than we can tackle the slave drivers.

**Comment by tbursztyka** [comment]
Let's try this quick fix for cc2520, hopefully that will pass now.

File: dts/bindings/spi/intel,intel-spi.yaml (hunk 1 above)
Comment (galak) [reply dts/bindings/spi/intel,intel-spi.yaml]: Ok, maybe we should but a bit more description  in 'description:' field about what SoC this applies to right now.

**Review by nashif — APPROVED** [review]

File: drivers/ieee802154/Kconfig.cc2520 [hunk 2]
```diff
@@ -14,18 +14,21 @@ if IEEE802154_CC2520
 
 config IEEE802154_CC2520_DRV_NAME
 	string "TI CC2520 Driver's name"
+	depends on !HAS_DTS_SPI
```
Comment (galak) [inline drivers/ieee802154/Kconfig.cc2520]: Under what situation after this PR will be able to set !HAS_DTS_SPI?  Seems like you should just remove this.

File: boards/x86/quark_se_c1000_devboard/quark_se_c1000_devboard.dts [hunk 3]
```diff
@@ -78,4 +78,5 @@
 
 &spi1 {
 	status = "ok";
+
```
Comment (galak) [inline boards/x86/quark_se_c1000_devboard/quark_se_c1000_devboard.dts]: Is this missing the cc2520 node?

File: dts/bindings/spi/intel,intel-spi.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+# Copyright (c) 2018 Synopsys, Inc. All rights reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+---
+title: Intel SPI Controller
+version: 0.1
+
+description: >
+     This is a representation of the Intel SPI controller.
```
Comment (galak) [inline dts/bindings/spi/intel,intel-spi.yaml]: We need to add a bit more than just 'Intel SPI Controller' so someone has a bit of clue when to use this binding/compatible.

File: dts/x86/intel_curie.dtsi [hunk 5]
```diff
@@ -166,7 +166,7 @@
 				reg = <0x0>;
 				label = "cc2520";
 				spi-max-frequency = <8000000>;
-				status = "disabled";
+				status = "ok";
```
Comment (galak) [inline dts/x86/intel_curie.dtsi]: Why this change here?  Typically done in the board.dts

**Review by galak — CHANGES_REQUESTED** [review]
* Some DTS cleanup/changes
* Also not sure what's going on w/the changes to the samples/drivers/led_ws2812 and samples/drivers/led_lpd8806?

File: dts/bindings/spi/intel,intel-spi.yaml (hunk 4 above)
Comment (tbursztyka) [reply dts/bindings/spi/intel,intel-spi.yaml]: Why should this binding be more precise while other are exactly the same? (I can add galileo in the mix, but that's not really the place)

**Comment by tbursztyka** [comment]
@galak about the led_strips samples, these were uselessly renaming spi controller name in order no to some irrelevant "double check" on spi controller being up, when it's all about led strip (if it's badly configured, the sample won't work and that's it).

File: dts/x86/intel_curie.dtsi (hunk 5 above)
Comment (tbursztyka) [reply dts/x86/intel_curie.dtsi]: that's due to a wrong rebase, move it to proper commit

**Review by dbkinder — APPROVED** [review]
doc changes LGTM

**Comment by tbursztyka** [comment]
@galak pleate, re-review.

File: dts/x86/intel_curie.dtsi (hunk 5 above)
Comment (galak) [reply dts/x86/intel_curie.dtsi]: ? shouldn't the addition of the cc2520@0 node & cs-gpios be in the board.dts.  Might still be a rebase issue.

Comment (tbursztyka) [reply dts/x86/intel_curie.dtsi]: no you'r right, cc2520 is present on the board, not on the soc

**Comment by tbursztyka** [comment]
@galak  So you forked only part of the PR? That's even more not a proper way. What is this process of working ...

**Comment by galak** [comment]
> @galak So you forked only part of the PR? That's even more not a proper way. What is this process of working ...

We'll it annoys me that github doesn't let me merge parts of a PR, so from time to time I do this so I can stop reviewing the same commits over and over again.  Sorry for not explain that in more detail.
